### PR TITLE
Update MacOS packaging system for signing and notarization

### PR DESCRIPTION
**Pull request by neuroscr**
and
- always clean out old binary (in case of failure)
- use `make release` binary paths
- generate distribution.xml, so it no longer needs to be updated separately to bump the version
- adds lokinetctl to packaging

### CODE REVIEW

**Comment by neuroscr** [comment]
MacOS already has libcurl as part of the system. I believe libsodium is statically linked in this case but I'll double check

File: osx-setup/make-release.sh [hunk 1]
```diff
@@ -16,6 +22,20 @@ cp /usr/local/lib/libuv.dylib osx-pkg/usr/local/lib
 # just incase they want to switch networks later
 rm osx-pkg/usr/local/bin/lokinet-bootstrap
 cp ../lokinet-bootstrap osx-pkg/usr/local/bin
+#echo "Copying lokinet-control-panel.app"
+rm -fr osx-pkg/Applications/Lokinet.app
+# requires -R to keep all the symbolic links inside the App
+# https://blog.inventic.eu/2015/03/os-x-codesign-failed-bundle-format-is-ambiguous-could-be-app-or-framework/
+cp -R Lokinet.app osx-pkg/Applications
+if [ ! -f "lokinet-brand_icon-only.icns" ]; then
+  curl http://installer.lokinet.org/lokinet-brand_icon-only.icns > lokinet-brand_icon-only.icns
```
Comment (majestrate) [inline osx-setup/make-release.sh]: but why?

Comment (neuroscr) [reply osx-setup/make-release.sh]: thought you didn't want binary files in the repo

Comment (majestrate) [reply osx-setup/make-release.sh]: why http?

Comment (neuroscr) [reply osx-setup/make-release.sh]: because I'm not concerned about someone hacking our icon

File: osx-setup/make-release.sh [hunk 2]
```diff
@@ -3,60 +3,72 @@ VERSION=$1
...
 #pkgutil --check-signature lokinet_macos64_installer_v$VERSION-signed.pkg
+
+echo "Configuring gatekeeper..."
 spctl -a -v --type install lokinet_macos64_installer_v$VERSION-signed.pkg
+
+echo "Submitting Notarization..."
 xcrun altool --notarize-app --primary-bundle-id "network.loki.loki-network.pkg" --username $EMAIL --password $PW --asc-provider "SUQ8J2PCT7" --file lokinet_macos64_installer_v$VERSION-signed.pkg
-echo "xcrun altool --notarization-info X --username $EMAIL --password $PW"
+
+echo "xcrun altool --notarization-info XXXXXXXXX --username $EMAIL --password $PW"
```
Comment (majestrate) [inline osx-setup/make-release.sh]: why are we echoing secrets to stdout?

Comment (despair86) [reply osx-setup/make-release.sh]: macOS a shit perhaps? even `signtool(1)` doesn't do this

**Comment by majestrate** [comment]
On Fri, Jan 10, 2020 at 04:45:10AM -0800, Rick V wrote:
> despair86 commented on this pull request.
> 
> 
> 
> >  xcrun altool --notarize-app --primary-bundle-id "network.loki.loki-network.pkg" --username $EMAIL --password $PW --asc-provider "SUQ8J2PCT7" --file lokinet_macos64_installer_v$VERSION-signed.pkg
> -echo "xcrun altool --notarization-info X --username $EMAIL --password $PW"
> +
> +echo "xcrun altool --notarization-info XXXXXXXXX --username $EMAIL --password $PW"
> 
> macOS a shit perhaps? even `signtool(1)` doesn't do this

no, this script is echoing --password $PW as an infirmational message which is.... wat

File: .github/workflows/ccpp.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v1
+    - name: mkdir
+      run: mkdir build && cd build && cmake ..
+    - name: make
+      run: make
```
Comment (majestrate) [inline .github/workflows/ccpp.yml]: `make -C build` technically

Comment (neuroscr) [reply .github/workflows/ccpp.yml]: I forgot this was opened against my master, probably should back this out

File: osx-setup/make-release.sh (hunk 2 above)
Comment (neuroscr) [reply osx-setup/make-release.sh]: It’s a parameter passed in, so it’s already in the console. It’s just echo out shortcuts of commands you have to run manually after (I haven’t been able to programmatically capture the output and parse it yet)

**Comment by majestrate** [comment]
On Thu, Jan 16, 2020 at 03:11:49AM -0800, Ryan Tharp wrote:
> neuroscr commented on this pull request.
> 
> 
> 
> >  xcrun altool --notarize-app --primary-bundle-id "network.loki.loki-network.pkg" --username $EMAIL --password $PW --asc-provider "SUQ8J2PCT7" --file lokinet_macos64_installer_v$VERSION-signed.pkg
> -echo "xcrun altool --notarization-info X --username $EMAIL --password $PW"
> +
> +echo "xcrun altool --notarization-info XXXXXXXXX --username $EMAIL --password $PW"
> 
> It’s a parameter passed in, so it’s already in the console. It’s just echo out shortcuts of commands you have to run manually after (I haven’t been able to programmatically capture the output and parse it yet)

my point is that it shouldn't even do that

**Pull request closed without merging**